synapse/rest/client/v1/room.py

```diff
@@ -222,15 +224,19 @@ def register(self, http_server):
     async def on_POST(self, request, room_id, event_type, txn_id=None):
         requester = await self.auth.get_user_by_req(request, allow_guest=True)
         content = parse_json_object_from_request(request)
+        prev_events = parse_strings_from_args(request.args, "prev_event")
 
         event_dict = {
             "type": event_type,
             "content": content,
             "room_id": room_id,
-            "sender": requester.user.to_string(),
+            "sender": requester.user.to_string()
         }
 
-        if b"ts" in request.args and requester.app_service:
+        if prev_events:
+            event_dict["prev_events"] = prev_events
+
+        if b"ts" in request.args: # and requester.app_service:
```

Comment:
Commented this out just so it's easier for testing in Complement.

TODO: Need to figure out how to define myself as an appservice in my Complement MR: https://github.com/matrix-org/complement/pull/68

Reply:
I think this needs to be done before merging?

Reply:
Yes! But I haven't done yet because then the Complement tests won't run. See the TODO above about making Complement run as an appservice so that this logic will work here.